### PR TITLE
(CDAP-12765) Allow service http server resources settable

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/SystemArguments.java [hunk 1]
```diff
@@ -278,6 +292,26 @@ public static Resources getResources(Map<String, String> args, @Nullable Resourc
   }
 
   /**
+   * Configures a {@link NettyHttpService.Builder} based on the give arguments.
```
Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/SystemArguments.java]: s/give/given

File: cdap-docs/developer-manual/source/building-blocks/services.rst [hunk 2]
```diff
@@ -211,11 +211,29 @@ If both the memory and the number of cores needs to be set, this can be done usi
...
 An example of setting ``Resources`` using runtime arguments is shown in :ref:`Purchase
 <examples-purchase>` example's ``PurchaseHistoryBuilder.java``.
 
 .. _services-routing:
 
+Service Thread Model
+====================
+An HTTP server is started for each Service instance, which by default starts 60 threads to handle
+client requests. Each thread is basically tied to one active client request and each thread would
+have its own instances of ``HttpServiceHandler``\s. This guarantees there will be no concurrent
```
Comment (anwar6953) [inline cdap-docs/developer-manual/source/building-blocks/services.rst]: s/instances/instance

**Comment by anwar6953** [comment]
Two minor comments; remainder LGTM.

**Comment by chtyim** [comment]
Build passed https://builds.cask.co/browse/CDAP-DUT6175

Push and squash a commit that fixes the docs

**Pull request merged**